src/lsp/cobol_lsp/lsp_document_symbol.ml

```diff
@@ -0,0 +1,338 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                        SuperBOL OSS Studio                             *)
+(*                                                                        *)
+(*  Copyright (c) 2022-2023 OCamlPro SAS                                  *)
+(*                                                                        *)
+(* All rights reserved.                                                   *)
+(* This source code is licensed under the GNU Affero General Public       *)
+(* License version 3 found in the LICENSE.md file in the root directory   *)
+(* of this source tree.                                                   *)
+(*                                                                        *)
+(**************************************************************************)
+
+open Lsp.Types
+open Cobol_common.Srcloc.INFIX
+open Lsp.Types.DocumentSymbol
+open Cobol_ptree
+module Visitor = Cobol_common.Visitor
+type range = Range.t
+type doc_symbol = t
+
+let enclosing_range (range: range) (buffer: doc_symbol list)  =
+  let end_ = match buffer with
+    | [] -> range.end_
+    | last::_ -> last.range.end_
+  in
+  Range.create ~start:range.start ~end_
+
+let create ~range ?(children=[]) =
+  create ~range ~selectionRange:range ~children:(List.rev children)
+
+type proc_acc =
+  {
+    result: doc_symbol list;
+    prev_section: (range * string) option;
+    buffer: doc_symbol list;
+  }
+let proc_init_acc = { result = []; prev_section = None; buffer = [] }
+
+let complete_section { result; prev_section; buffer=children } =
+  match prev_section with
+  | None -> children @ result
+  | Some (range, name) ->
+    let range = enclosing_range range children in
+    create ~kind:Function ~name ~range ~children ()
+    :: result
+
+let proc_folder range_from: proc_acc Cobol_ptree.Visitor.folder =
```

Comment:
```suggestion
let proc_folder range_from : proc_acc Cobol_ptree.Visitor.folder =
```
(in this case the space clarifies that this is a function; otherwise one may think the type annotation applies to `range_from`).